pkg/backup/snapshotter/snapshotter_aws.go

```diff
@@ -102,15 +103,12 @@ func (s *AWSSnapshotter) AddVolumeTags(pvs []*corev1.PersistentVolume) error {
 	resourcesTags := make(map[string]util.TagMap)
 
 	for _, pv := range pvs {
-		podName := pv.GetAnnotations()[label.AnnPodNameKey]
-		pvcName := pv.GetName()
-		volId := pv.Spec.CSI.VolumeHandle
-
 		tags := make(map[string]string)
-		tags[PVCTagKey] = pvcName
-		tags[PodTagKey] = podName
+		tags[PVCTagKey] = pv.GetName()
+		tags[PodTagKey] = pv.GetAnnotations()[label.AnnPodNameKey]
+		tags[PodNSTagKey] = pv.GetAnnotations()[label.NamespaceLabelKey]
 
-		resourcesTags[volId] = tags
+		resourcesTags[pv.Spec.CSI.VolumeHandle] = tags
```

Comment:
you can get volume id by `pv.Annotations[constants.AnnRestoredVolumeID]`. `pv.Spec.CSI.VolumeHandle` may be empty. you can see `Snapshotter.GetVolumeID` and `Snapshotter.SetVolumeID` methods.

Reply:
done